spec/lib/pickup_bot/commands/leave_game_spec.rb

```diff
@@ -0,0 +1,74 @@
+require "spec_helper"
+require 'telegram/bot'
+require 'pickup_bot'
+
+feature "leaving a game" do
+  let(:bot) { Telegram::Bot::Client.new('fake-token') }
+  let(:pickup_bot) { PickupBot.new(bot) }
+  let(:telegram_user) { Telegram::Bot::Types::User.new(user_params) }
+  let(:telegram_chat) { Telegram::Bot::Types::Chat.new(chat_params) }
+  let(:existing_game) { Game.new(chat_id: fake_chat_id) }
+
+  before :each do
+    stub_request(:any, /api.telegram.org/).to_return(status: 200, body:"[]", :headers => {})
+  end
+
+  context "no game currently exists" do
+    scenario "user tries to leave the non-existant game" do
+      message = Telegram::Bot::Types::Message.new(message_params('/leave'))
+
+      pickup_bot.run(message)
+
+      expect(a_request(:post, "https://api.telegram.org/botfake-token/sendMessage").
+              with(body: {
+                "chat_id" => "123",
+                "text" => I18n.t(
+                  "bot.commands.leave.no_game",
+                  username: user_params[:username]
+                )
+              }
+            )).to have_been_made.times(1)
+    end
+  end
+
+  context "an active game exists" do
+    scenario "user tries to leave a game he's not attending" do
+      message = Telegram::Bot::Types::Message.new(message_params("/leave"))
+      game = Game.create(chat_id: fake_chat_id, required_players: 5)
+
+      pickup_bot.run(message)
+
+      expect(a_request(:post, "https://api.telegram.org/botfake-token/sendMessage").
```

Comment:
Since you're using it a lot, it might be easy to store the telegram url in a variable